src/pages/Embeds/Plio.vue

```diff
@@ -399,49 +400,32 @@ export default {
   methods: {
     ...mapActions("auth", ["setAccessToken", "setActiveWorkspace"]),
     /**
-     * Update the number of correct answers
-     * @param  {Number} itemIndex -  The index of the item to be considered for the calculation
-     * @param  {String, Number, Object} userAnswer - User's answer to that item
+     * Checks whether the item at the provided itemIndex is a subjective question
+     * and if the user has answered the question or not
+     * @param {Number} itemIndex - The index of the item in question
+     * @param {String, Number, Object} userAnswer - User's answer to that item
      */
-    updateNumCorrect(itemIndex, userAnswer) {
-      if (this.isItemMCQ(itemIndex)) {
-        const correctAnswer = this.items[itemIndex].details.correct_answer;
-        if (!isNaN(userAnswer) && userAnswer == correctAnswer) {
-          this.numCorrect += 1;
-        }
-      } else if (
-        this.isItemSubjective(itemIndex) &&
-        userAnswer != null &&
-        userAnswer.trim() != ""
-      ) {
-        this.numCorrect += 1;
-      }
-      return;
+    isSubjectiveQuestionAnswered(itemIndex, userAnswer) {
+      return (
+        this.isItemSubjective(itemIndex) && userAnswer != null && userAnswer.trim() != ""
+      );
     },
     /**
-     * Update the number of wrong answers
+     * Update the number of correct answers, wrong answers and skipped items
      * @param  {Number} itemIndex -  The index of the item to be considered for the calculation
      * @param  {String, Number, Object} userAnswer - User's answer to that item
      */
-    updateNumWrong(itemIndex, userAnswer) {
+    updateNumCorrectWrongSkipped(itemIndex, userAnswer) {
       if (this.isItemMCQ(itemIndex)) {
         const correctAnswer = this.items[itemIndex].details.correct_answer;
-        if (!isNaN(userAnswer) && userAnswer != correctAnswer) {
-          this.numWrong += 1;
+        if (!isNaN(userAnswer)) {
+          userAnswer == correctAnswer ? (this.numCorrect += 1) : (this.numWrong += 1);
+          // reduce numSkipped by 1 if numCorrect or numWrong increases
+          this.numSkipped -= 1;
```

Comment:
what if a question is already answered and then, I click the "skip button" - it will cause the scorecard metrics to be re-calculated and hence, the skip value will be reduced.